### ISSUE TITLE
Simple usage tutorial

### ISSUE THREAD

**Issue by tiagofilipe12** [bionode/bionode-waterwheel#49]
What do you think if we create a simple usage tutorial for some of the basic usages of tasks and orchestrators to README? We already have 4 examples but I think a walkthrough tutorial could help users to develop new pipelines.

**Comment by thejmazz** [comment]
Yes, this is definitely an important task if we wish to attract newcomers more easily.

I think a thorough walkthrough might include:
- go through a short pipeline by hand (should be copy-pastable commands but still get point across that files are written/read across multiple programs)
- show makefile/snakemake/nextflow and how they suffer when using multiple alternate params/programs for a pipeline step
- show how watermill overcomes that (i.e reusable tasks across pipelines, fork)

Maybe we should have a "why" tutorial (i.e. the points above) as well as a "getting started" one. The why one would go through pitfalls of other tools/practices which explains the motivation for watermill. The getting started one explains how the pipeline DAG works and how to create tasks using

- shell commands
- JS functions
- streams

**Comment by tiagofilipe12** [comment]
PR [55](https://github.com/bionode/bionode-watermill/pull/55) should have handled the "getting started" tutorial as well as [bionode-watermill-tutorial](https://github.com/bionode/bionode-watermill-tutorial). 
Regarding the "why", we could put a link to your [blog](https://jmazz.me/blog/NGS-Workflows) in README as well as the "getting started" tutorial link.
